webviews/common/context.tsx

```diff
@@ -176,6 +176,13 @@ export class PRContext {
 
 	public openDiff = (comment: IComment) => this.postMessage({ command: 'pr.open-diff', args: { comment } });
 
+	public toggleResolveComment = (reviewThread: ReviewEvent['reviewThread'], thread: IComment[]) => {
+		if (!reviewThread) {
+			return;
+		}
+		this.postMessage({ command: 'pr.resolve-comment-thread', args: { threadId: reviewThread.threadId, toResolve: reviewThread.isResolved, thread } });
```

Comment:
Might need to await this `postMessage` and then call `updatePR`. That could be where the unexpected UI behavior we were seeing is coming from.

Reply:
Actually I was only missing the `updatePR` call. I was changing the values in the timeline but I did not update the state which caused the inconsistent state.